File: test-automation-core/src/main/java/com/nortal/test/core/util/PrettyPrintUtils.java
package com.nortal.test.core.util;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonSyntaxException;

/**
 * Utils for pretty printing objects as json
 */
public final class PrettyPrintUtils {

	private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();

	private PrettyPrintUtils() {
		//It's a static utils class
	}

	public static String prettyPrint(Object object) {
		try {
			return "\n" + GSON.toJson(object, Object.class);
		} catch (JsonSyntaxException e) {
			return object.toString();
		}
	}


}

File: test-automation-core-old/src/main/java/com/nortal/test/core/util/PrettyPrintUtils.kt
package com.nortal.test.core.util

import com.google.gson.GsonBuilder
import com.google.gson.JsonSyntaxException

/**
 * Utils for pretty printing objects as json
 */
object PrettyPrintUtils {
    private val GSON = GsonBuilder().setPrettyPrinting().create()
    fun prettyPrint(`object`: Any): String {
        return try {
            """
     
     ${GSON.toJson(`object`, Any::class.java)}
     """.trimIndent()
        } catch (e: JsonSyntaxException) {
            `object`.toString()
        }
    }
}